SPLAT-1158: platform external e2e workflow using AWS as provider

mtulio opened the pull request:
Introduce e2e/conformance workflow and jobs to install OpenShift on a well-known provider (AWS) with platform type `External` with and without CCM, using default OpenShift conformance workflow (openshift-e2e) and OPCT.

The workflow is implementing the steps provided in "[Installing a cluster with Platform External](https://docs.providers.openshift.org/platform-external/installing/)" from the "Infrastructure Provider Onboarding Guide" (IPOG).

Summary of items added:

- Step 1 of Platform External guide from IPOG: generate config
- Step 2 of Platform External guide from IPOG: provision infrastructure
- Step 3 of Platform External guide from IPOG: deploy CCM on temporary control plane (bootstrap)
- UPI install workflow/waiter
- UPI modified cloudformation templates (follow up PRs will move to CloudFormation stackset)
- e2e workflow for external-aws + ccm
- opct conformance workflow for external-aws-ccm
- jobs on installer and origin repo
- skip list is added to prevent workflow failures (short-term)
- origin repo isn't skipping tests to fix middle-term failures

## Job Executions (Highlights)

Samples, see more at [PR history](https://prow.ci.openshift.org/pr-history/?org=openshift&repo=release&pr=45921):

[j1] CI E2E flow for OCP Platform External on AWS (w/o CCM):
- [nightly-4.15-e2e-external-aws #1757101383682625536](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/45921/rehearse-45921-periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws/1757101383682625536)

[j2] CI E2E flow for OCP Platform External on AWS with CCM:
- [nightly-4.15-e2e-external-aws-ccm #1757101483590946816](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/45921/rehearse-45921-periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm/1757101483590946816)
- [nightly-4.15-e2e-external-aws-ccm #1756928258894467072](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/45921/rehearse-45921-periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm/1756928258894467072)

[j3] OPCT E2E flow for OCP Platform External on AWS (w/o CCM):
- [nightly-4.15-opct-external-aws #1757255745801293824](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/45921/rehearse-45921-periodic-ci-openshift-release-master-nightly-4.15-opct-external-aws/1757255745801293824)

[j4] OPCT E2E flow for OCP Platform External on AWS with CCM:
- [nightly-4.15-opct-external-aws-ccm #1757132567846326272](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/45921/rehearse-45921-periodic-ci-openshift-release-master-nightly-4.15-opct-external-aws-ccm/1757132567846326272)
- [nightly-4.15-opct-external-aws-ccm #1757495441294888960](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/45921/rehearse-45921-periodic-ci-openshift-release-master-nightly-4.15-opct-external-aws-ccm/1757495441294888960)
- [nightly-4.15-opct-external-aws-ccm #1757267257693573120](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/45921/rehearse-45921-periodic-ci-openshift-release-master-nightly-4.15-opct-external-aws-ccm/1757267257693573120)

## Notes

The outcome can feed/contribute to the following work items:

- Install Platform External on AWS with CCM to increase coverage:
  - https://issues.redhat.com/browse/SPLAT-1158
  - https://issues.redhat.com/browse/OCPCLOUD-1911

- Platform External install workflow improvement:
    - https://issues.redhat.com/browse/SPLAT-1204
    - https://issues.redhat.com/browse/SPLAT-1196
    - https://issues.redhat.com/browse/SPLAT-1191
    - https://issues.redhat.com/browse/SPLAT-1190

- Infrastructure Onboarding Guide / Platform External documentation and use case:
  - https://issues.redhat.com/browse/SPLAT-1147
  - https://issues.redhat.com/browse/SPLAT-1060
  - https://issues.redhat.com/browse/OCPCLOUD-1581

- Platform External e2e research: 

- OPCT on AWS with Platform External (None+CCM), and PLatform None to feed the baseline for VCSP: https://issues.redhat.com/browse/OPCT-231
- validation of behavior of generics upstream e2e to validate service load balancers: https://issues.redhat.com/browse/SPLAT-1224

## Check list

- [x] validate platform type external provisioning on AWS for regular e2e workflow
- [x] validate platform type external provisioning on AWS for regular e2e workflow with CCM
- [x] validate platform type external provisioning on AWS for default OPCT conformance workflow
- [x] validate platform type external provisioning on AWS for default OPCT conformance workflow with CCM
- [x] validate  platform type external presubmit on origin repo (expect to fail as we are enforcing no e2e skips)
- [x] validate  platform type external presubmit on installer

## Isolated PRs:

- [x] https://github.com/openshift/release/pull/48980
- [x] https://github.com/openshift/release/pull/52026


## Approval matrix:

| ✅  | change group | Team/owner | Status |
| -- | -- | -- | -- |
| ✅ ✔️  | [e2e workflow][e2e] | splat/cloud | splat([lgtm][lgtm-splat]) cloud([ok/unhold][lgtm-cloud]) |
| ✅ | [opct conformance workflow][opct] | splat |splat([lgtm][lgtm-splat])  |
| ✅✔️  | [IPOG platform external steps][ipog] | splat/cloud |splat([lgtm][lgtm-splat]) cloud([ok/unhold][lgtm-cloud]) |
| ✅ | presubmit [master][inst-pre], [4.16][inst-pre416] | installer | [approved][ap-inst] |
| 🚧 | presubmit [origin@master][orig-pre] | [owners][owners] | [Waiting for review][trt-ask] presubmits |
| 🚧| periodics [nightly][per-nightly], [ci][per-ci] | TRT ([owners][owners]) | [Waiting for review][trt-ask] periodics |

[owners]: https://github.com/openshift/release/pull/45921#issuecomment-2113098255
[trt-ask]: https://github.com/openshift/release/pull/45921#issuecomment-2113175781
[lgtm-splat]: https://github.com/openshift/release/pull/45921#issuecomment-2112349801
[ap-inst]: https://github.com/openshift/release/pull/45921#pullrequestreview-2058636158
[slack-cloud]: https://redhat-internal.slack.com/archives/C02DSMAP4LA/p1715779593361909
[lgtm-cloud]: https://github.com/openshift/release/pull/45921#pullrequestreview-2110991521

[inst-pre]: https://github.com/openshift/release/pull/45921/files#diff-92fb8761a3401342e7a5678fa916ff7f552fe64f9fa412105c354e691c76a736
[inst-pre416]: https://github.com/openshift/release/pull/45921/files#diff-9c81f51dc1c38159a5d3116bd540ace36e73fdb4a612e8481861a64a5eabad7e

[per-nightly]: https://github.com/openshift/release/pull/45921/files#diff-402cf6f3b671603791772b54dfe8ac7bc2f459edf6d3680057fc54a7b39c1d8d
[per-ci]: https://github.com/openshift/release/pull/45921/files#diff-8fa01be75dd1292c360eb324d72a08c671bcb1ee7a15a8040cc450fd0b956547

[orig-pre]: https://github.com/openshift/release/pull/45921/files#diff-896f64fc999b47a8bf87e75663d57fc9093909a645b0abaeb2371bf4179b133e

[e2e]: https://github.com/openshift/release/pull/45921/files#diff-ac552a6507719ccab58a6aa300dcc02d0f754e515081ce97f66dc897631b9cd4
[opct]: https://github.com/openshift/release/pull/45921/files#diff-17bbed49b9dcd358cf73de02a6ede4f21bbfd98603d81af400d7dcfa8ad4962f
[ipog]: https://github.com/openshift/release/pull/45921/files#diff-2346d9d60df807ca52aa3a330faeab731350e5e04442be54221dc41639d8c491

mtulio commented:
/test all

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/test all

mtulio commented:
/test all

mtulio commented:
/test core-valid

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
> /pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

CloudFormation (external) upgrade error, fixed, retrying: 

/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm



> @mtulio: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> 
> Test name	Commit	Details	Required	Rerun command
> ci/rehearse/periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm	[3b9ff7f](https://github.com/openshift/release/commit/3b9ff7f352fd07cd389c32b4e55212d53b0bf466)	[link](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/45921/rehearse-45921-periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm/1727147921566076928)	unknown	`/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm`
> ci/prow/core-valid	[da7545e](https://github.com/openshift/release/commit/da7545efcbc176821b74ebd2672c8bce040047d6)	[link](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/45921/pull-ci-openshift-release-master-core-valid/1726994241034391552)	true	`/test core-valid`
> [Full PR test history](https://prow.ci.openshift.org/pr-history?org=openshift&repo=release&pr=45921). [Your PR dashboard](https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:mtulio).

us-west-1 does not have 3 zones. Setting VPC stack to use two zones


/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
> /pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

API didn't come up.

/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
> /pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

4.15 nightly is crashing, connecting to one master I can see OVN DB issues:

~~~
[systemd]
Failed Units: 2
  ovsdb-server.service
  systemd-sysusers.service
~~~

Adding a periodic for 4.14 to try it again in a healthy release

/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-external-aws

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ec-4.15-e2e-external-aws-ccm

mtulio commented:
/test all

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ec-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ec-4.15-e2e-external-aws

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ec-4.15-e2e-external-aws

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ec-4.15-e2e-external-aws

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ec-4.15-e2e-external-aws

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ec-4.15-e2e-external-aws

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ec-4.15-e2e-external-aws

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ec-4.15-e2e-external-aws

mtulio commented:
The default image set by the step (and installer binary) was the cause of cluster failures on boot, forcing the image override to an 4.15-EC make progress in the provisioning. The last run passed in the stages of provisioning and post-config, failing in the bootstrap waiter for typos. Retrying targeting to reach the stage of e2e:

/pj-rehearse periodic-ci-openshift-release-master-ec-4.15-e2e-external-aws

/pj-rehearse periodic-ci-openshift-release-master-ec-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ec-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ec-4.15-e2e-external-aws

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ec-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
One important note of the OCI failures (nodes not joining to the cluster) is the size of hostname higher than allowed. This is the kubelet logs:
~~~
Nov 29 01:23:06 inst-4pewn-ci-op-h5011lsk-a68c7-master.private.oci.oraclevcn.com kubenswrapper[5644]: E1129 01:23:06.072714    5644 kubelet_node_status.go:94] "Unable to register node with API server" err="Node \"inst-4pewn-ci-op-h5011lsk-a68c7-master.private.oci.oraclevcn.com\" is invalid: metadata.labels: Invalid value: \"inst-4pewn-ci-op-h5011lsk-a68c7-master.private.oci.oraclevcn.com\": must be no more than 63 characters" node="inst-4pewn-ci-op-h5011lsk-a68c7-master.private.oci.oraclevcn.com"
~~~

Adjusting the instance pool with shorter hostname (terraform) resolved for CI.

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-oci-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ec-4.15-e2e-external-aws

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/test all

mtulio commented:
/test all

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/test all

mtulio commented:
/test all

mtulio commented:
/test all

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/test all

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-opct-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-opct-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-opct-external-aws-ccm

mtulio commented:
/test all

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-opct-external-aws

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-opct-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-opct-external-aws-ccm

mtulio commented on ci-operator/step-registry/platform-external/ccm/conf/aws/platform-external-ccm-conf-aws-commands.sh:
```diff
@@ -0,0 +1,145 @@
+
+# Build from: https://github.com/openshift/cloud-provider-aws/blob/master/Dockerfile.openshift
+CCM_IMAGE="quay.io/mrbraga/openshift-cloud-provider-aws:latest"
+CCM_NAMESPACE=openshift-cloud-controller-manager
```
can we use as a env var? Shouldn't we use NS with prefix `openshift`?

mtulio commented on ci-operator/step-registry/platform-external/ccm/conf/aws/platform-external-ccm-conf-aws-commands.sh:
```diff
@@ -0,0 +1,145 @@
+
+echo_date "Creating CloudController Manager deployment"
+
+cat << EOF | envsubst > $CCM_MANIFEST_PATH
```
Should we need to create the "base" deployment file, then in later steps patch with customizations for each provider?
Example steps:
- `platform-external-ccm-conf`: creates the deployment
- `platform-external-ccm-conf-aws`:  patch the deployment with specific config (like image, command, ports, etc)

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-opct-external-aws-ccm

mtulio commented:
/test all

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-opct-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-opct-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ci-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ci-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ci-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ci-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ci-4.15-e2e-external-aws-ccm

mtulio commented:
The flow can be installed without forcing the release image outside CI. All greens executions I had in the history were when forcing the release image to RC, using nighties or CI ones impact in perm failures, not sure why, still investigating. My next change will return the hard coded release image to RC to make sure any recent changes is also impacted (as I needed to change the base image when enabling CI job).

mtulio commented on ci-operator/step-registry/platform-external/cluster/aws/install/platform-external-cluster-aws-install-commands.sh:
```diff
@@ -0,0 +1,450 @@
+#!/bin/bash
```
TODO check if we can move this to a generic upi flow (currently it has aws type bits)

mtulio commented:
> The flow can be installed without forcing the release image outside CI. All greens executions I had in the history were when forcing the release image to RC, using nighties or CI ones impact in perm failures, not sure why, still investigating. My next change will return the hard coded release image to RC to make sure any recent changes is also impacted (as I needed to change the base image when enabling CI job).


/pj-rehearse periodic-ci-openshift-release-master-ci-4.15-e2e-external-aws-ccm

mtulio commented:
> /pj-rehearse periodic-ci-openshift-release-master-ci-4.15-e2e-external-aws-ccm

Fixing remaining issues in the change image from `upi-installer`

/pj-rehearse periodic-ci-openshift-release-master-ci-4.15-e2e-external-aws-ccm

mtulio commented:
rollback to installer image in UPI step

/pj-rehearse periodic-ci-openshift-release-master-ci-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ci-4.15-e2e-external-aws-ccm

mtulio commented:
returning to nightlies - lot of issues related to dependencies with CI job

/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/test all

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse pull-ci-openshift-installer-master-e2e-external-aws

mtulio commented:
/reopen
/remove-lifecycle rotten

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws

mtulio commented:
/test all

mtulio commented:
/test pull-ci-openshift-release-master-build-clusters
/test release-controller-config

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws

mtulio commented:
Observing registry login errors on bootstrap:

> Apr 30 21:02:16 ip-10-0-7-238 release-image-download.sh[98196]: Error: initializing source docker://registry.build05.ci.openshift.org/ci-op-i3c7i65b/release@sha256:0d4f2927919960d8945941c616e8755d77f2d1012c72a75c5b331255acb13c15: reading manifest sha256:0d4f2927919960d8945941c616e8755d77f2d1012c72a75c5b331255acb13c15 in registry.build05.ci.openshift.org/ci-op-i3c7i65b/release: authentication required


Trying to append the registry secret to pull secret:

/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws

mtulio commented:
CSR approval for compute nodes is failing. Trying a new version...

/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws

mtulio commented:
e2e executed correctly, with expected test failures that need to be addressed to the test core.

Testing again the external w/o CCM (default, above) and CCM:

/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws-ccm

mtulio commented:
> /pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws-ccm

Got successful installations - jobs failed due e2e bugs that need to be addressed (out of scope).

The CCM job wasn't enabling CCM through install-config. Fixed it and run new jobs after clean up:

/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws-ccm

mtulio commented:
> 
> Got successful installations - jobs failed due e2e bugs that need to be addressed (out of scope).
> 
> The CCM job wasn't enabling CCM through install-config. Fixed it and run new jobs after clean up:
> 
> /pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws-ccm

e2e result keeps consistent results (success install).
e2e-ccm is failing due missing pull secret with CI creds to extract CCM image from release payload. Trying again:

/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ci-4.15-opct-external-aws

mtulio commented:
> /pj-rehearse periodic-ci-openshift-release-master-ci-4.15-opct-external-aws

Isolate flake on waiting cluster when testing with opct workflow:

/pj-rehearse periodic-ci-openshift-release-master-ci-4.15-opct-external-aws

mtulio commented on ci-operator/step-registry/platform-external/cluster/aws/install/README.md:
```diff
@@ -0,0 +1,54 @@
+NÂO USAR
```
Remove

mtulio commented on ci-operator/step-registry/platform-external/cluster/wait-for/complete/platform-external-cluster-wait-for-complete-commands.sh:
```diff
@@ -0,0 +1,59 @@
+
+date "+%F %X" > "${SHARED_DIR}/CLUSTER_INSTALL_END_TIME"
+
+# touch ${ARTIFACT_DIR}/install-complete
```
Remove

mtulio commented on ci-operator/step-registry/platform-external/pre/conf/manifests/platform-external-pre-conf-manifests-commands.sh:
```diff
@@ -0,0 +1,138 @@
+            exit 0
+        fi
+
+        PROVIDER_ID=${PROVIDER_ID_COMMAND}
```
Can we add the command to discover the providerId as a field in install-config.yaml, making installer create this MachineConfig when platform is External, and CCM is enabled?

mtulio commented:
> > /pj-rehearse periodic-ci-openshift-release-master-ci-4.15-opct-external-aws
> 
> Isolate flake on waiting cluster when testing with opct workflow:
> 
> /pj-rehearse periodic-ci-openshift-release-master-ci-4.15-opct-external-aws

Fix OPCT workflow dependencies for CI image:

/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-opct-external-aws

mtulio commented:
> > > /pj-rehearse periodic-ci-openshift-release-master-ci-4.15-opct-external-aws
> > 
> > 
> > Isolate flake on waiting cluster when testing with opct workflow:
> > /pj-rehearse periodic-ci-openshift-release-master-ci-4.15-opct-external-aws
> 
> Fix OPCT workflow dependencies for CI image + adding 4.16 jobs for OPCT:
> 
> /pj-rehearse periodic-ci-openshift-release-master-ci-4.16-opct-external-aws

Not started:

/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-opct-external-aws

checking current required
/test all

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ci-4.15-opct-external-aws

mtulio commented:
Removing `upi-installer` as step dependency targeting to fix opct flow:


/pj-rehearse periodic-ci-openshift-release-master-ci-4.15-opct-external-aws

mtulio commented:
Testing regular e2e with opct fixes (upi-installer images removed):

/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws-ccm

mtulio commented:
OPCT finished successfully but now e2e is failing in install image when installing aws due missing pip (?). aws-install step wasnt changed in latest updates.

Check list:

| flow name | version | status | note |
| -- | -- | -- | -- |
| e2e-external-aws | nightly-4.16 | passing | 1785575083541008384 |
| e2e-external-aws-ccm | nightly-4.16 | failing  | installer image is missing pip |
| opct-external-aws | ci-4.15 | passing | 1786017439356555264 |
| opct-external-aws-ccm | TBD | TODO | -- |

Trying to fix installer image (awscli install) in e2e-ccm:

/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws-ccm

mtulio commented:
Check list:

| flow name | version | status | note |
| -- | -- | -- | -- |
| e2e-external-aws | nightly-4.16 | passing | 1785575083541008384 |
| e2e-external-aws-ccm | nightly-4.16 | failing  | installer image is missing pip |
| opct-external-aws | ci-4.15 | passing | 1786017439356555264 |
| opct-external-aws-ccm | TBD | TODO | -- |

Trying to fix installer image (awscli install) in e2e-ccm:

/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws-ccm

mtulio commented:
moving to `upi-installer`  image from `ocp` namespace.


/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws-ccm

mtulio commented:
fix the step image:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws-ccm

mtulio commented:
more images have been fixed in the destroy flow:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ci-4.15-opct-external-aws

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-external-vsphere-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws-ccm

mtulio commented on ci-operator/step-registry/platform-external/cluster/aws/install/platform-external-cluster-aws-install-commands.sh:
```diff
@@ -0,0 +1,421 @@
+# TODO(mtulio): move to step vars
+MACHINE_CIDR=10.0.0.0/16
+
+# begin bootstrapping
+if openshift-install coreos print-stream-json 2> "${ARTIFACT_DIR}/err.txt" > /tmp/coreos.json; then
+  RHCOS_AMI="$(jq -r --arg region "$AWS_REGION" '.architectures.x86_64.images.aws.regions[$region].image' /tmp/coreos.json)"
+  # if [[ "${CLUSTER_TYPE}" == "aws-arm64" ]] || [[ "${OCP_ARCH}" == "arm64" ]]; then
+  #   RHCOS_AMI="$(jq -r --arg region "$AWS_REGION" '.architectures.aarch64.images.aws.regions[$region].image' coreos.json)"
+  # fi
+else
+  RHCOS_AMI="$(jq -r --arg region "$AWS_REGION" '.amis[$region].hvm' /var/lib/openshift-install/rhcos.json)"
+fi
```
this can be moved to pre-steps as a long-term goal to prevent using `openshift-install` in installing for provider-specific steps.

mtulio commented on ci-operator/step-registry/platform-external/cluster/aws/pre/platform-external-cluster-aws-pre-chain.yaml:
```diff
@@ -0,0 +1,15 @@
+  - ref: platform-external-cluster-wait-for-api-bootstrap
+  - ref: platform-external-ccm-conf-aws
+  - ref: platform-external-ccm-deploy
+  - ref: platform-external-cluster-wait-for-ready-control
+  - ref: platform-external-cluster-wait-for-ready-compute
+  - ref: platform-external-cluster-wait-for-complete
```
maybe create a chain to summarize it:
```suggestion
  - chain: platform-external-cluster-wait
```

mtulio commented on ci-operator/step-registry/platform-external/cluster/aws/pre/platform-external-cluster-aws-pre-chain.yaml:
```diff
@@ -0,0 +1,15 @@
+  steps:
+  - chain: platform-external-cluster-pre
+  - ref: platform-external-cluster-aws-install
+  - ref: platform-external-cluster-wait-for-api-bootstrap
+  - ref: platform-external-ccm-conf-aws
+  - ref: platform-external-ccm-deploy
```
What about this chain including three steps?
```suggestion
  - chain: platform-external-ccm-conf-aws-deploy
```

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-external-aws-ccm

mtulio commented:
/test all

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws
/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws-ccm
/test ll

mtulio commented:
/test all

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws-ccm
/test all

mtulio commented:
fixes in the test skips, retrying:

/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws

mtulio commented:
increasing skip list

/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws

mtulio commented:
increasing skip list

/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws

mtulio commented:
The last ran does reflect the skip list, which failed with items in the list. Updating to ordered and uniq items:

/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws

mtulio commented:
Skip list is still wrong, ignoring tests: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/45921/rehearse-45921-periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws/1788002107249397760/artifacts/e2e-external-aws/openshift-e2e-test/build-log.txt

In the example above it was accepting only the last test from the list.

I fixed the parser to remove spaces between tests (`\|`). Let me try again:

/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws

mtulio commented:
adding double quote delimiter as openshift-tests still skipping some items from the list:

/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws

mtulio commented:
removing escape for double quote


/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws

mtulio commented:
removing escape for double quote, and disable monitor for LB:

/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws

mtulio commented:
test names reported in JUnit missmatch from the registered on openshift-tests, it was leading to ignore some values from test skips, specially for tests for some reason adding the word `[Jira:"kube-apiserver"]` in the name. Retesting:

/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws

mtulio commented:
> test names reported in JUnit missmatch from the registered on openshift-tests, it was leading to ignore some values from test skips, specially for tests for some reason adding the word `[Jira:"kube-apiserver"]` in the name. Retesting:
> 
> /pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws

More fixes related to the skip list.

/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws

mtulio commented:
The job is still failing. Investigating it looks like the e2e is not able to extract the `openshift-tests` from the image, but the credentials from build cluster is still there - appended in the install-config on install time, custom step added to the platform external (this PR).

Removing the pull-secret appending, and replacing creds used to discover CCM image. Trying a new shot:

/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws

mtulio commented:
restore registry login (bootstrap failed awaiting for CI creds), add skip for certs e2e which is ignoring pull secret:

/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws

mtulio commented on ci-operator/step-registry/openshift/e2e/external/aws/openshift-e2e-external-aws-workflow.yaml:
```diff
@@ -0,0 +1,28 @@
+      # https://issues.redhat.com/browse/OCPBUGS-29684
+      TEST_SKIPS: >-
+        \[sig-api-machinery\]\[Feature:ResourceQuota\] Object count check the quota after import-image with --all option \[Skipped:Disconnected\] \[Suite:openshift/conformance/parallel\]\|\[sig-arch\] ClusterOperators \[apigroup:config.openshift.io\] should define at least one namespace in their lists of related objects \[Suite:openshift/conformance/parallel\]\|\[sig-arch\] ClusterOperators \[api ...
+      # TODO(mtulio/elmiko): investigate why LB monitor test is triggered when platform type external without CCM.
```
FWIW @elmiko

mtulio commented:
> /pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws

PASS o/

Trying with CCM:

/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws-ccm

mtulio commented:
> > /pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws
> 
> PASS o/
> 
> Trying with CCM:
> 
> /pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws-ccm

deployment with CCM finished successfully. I just started the cleanups to have the final version. Updates:
- fix bootstrap destroy by setting the region when calling cloudformation
- moving RHCOS_AMI discovery to a dedicated step, pre-install, to remove `openshift-install` binary dependencies from install step
- move to env vars the instance types and machine CIDR in the install step
- force to clean the E2E_SKIPS to the origin presubmits, so we can work to fix the tests individually in follow up PRs
- moved the cloudformation templates to the Github org openshift-splat-team, repo `lab`: https://github.com/openshift-splat-team/labs/tree/platform-external-aws
- fix typo in logging

/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws-ccm

mtulio commented:
/pj-rehearse help

mtulio commented:
/pj-rehearse cancel

mtulio commented:
/pj-rehearse pull-ci-openshift-origin-master-e2e-external-aws

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws

mtulio commented:
/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws

mtulio commented:
/test all

mtulio commented:
/test all

mtulio commented:
/test release-controller-config

mtulio commented:
/test release-controller-config

mtulio commented:
/test release-controller-config

mtulio commented:
/test all

mtulio commented:
Preventing notifications until getting approval for stakeholders - I will tag each group based in the commits and component owners.

/uncc deads2k dgoodwin

Asking team/peer review:

/assign @rvanderp3 @elmiko @jcpowermac

Testing:
/pj-rehearse more

Holding until we resolve changes in https://github.com/openshift/release/pull/52026 (also added in this PR).
/hold

bradmwilliams commented on ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml:
```diff
@@ -56735,91 +56833,7 @@ periodics:
-      - mountPath: /secrets/ci-pull-credentials
-        name: ci-pull-credentials
-        readOnly: true
-      - mountPath: /usr/local/e2e-metal-ipi-bm-upgrade-cluster-profile
```
All the jobs that have `install-analysis-all` and/or `overall-analysis-all` in the name MUST be preserved

mtulio replied:
As recommended from Slack, the file `ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml` was restored from master and ran `make jobs` again. The `*-analysis-all` isnt removed. Thanks!

mtulio commented:
Starting final tests:

/pj-rehearse more

mtulio commented on ci-operator/config/openshift/installer/openshift-installer-master__altinfra.yaml:
```diff
@@ -510,6 +510,22 @@ tests:
       FEATURE_SET: CustomNoUpgrade
     workflow: openshift-e2e-vsphere-multi-vcenter
   timeout: 6h0m0s
+- always_run: false
+  as: e2e-external-aws
+  optional: true
+  steps:
+    cluster_profile: aws
+    workflow: openshift-e2e-external-aws
+  timeout: 6h0m0s
+- always_run: false
+  as: e2e-external-aws-ccm
+  optional: true
+  steps:
+    cluster_profile: aws
+    env:
+      PLATFORM_EXTERNAL_CCM_ENABLED: "yes"
+    workflow: openshift-e2e-external-aws
+  timeout: 6h0m0s
```
@r4f4 I am not sure if we'll need this jobs in `altinfra` as provisioning steps from this workflow is pure UPI. I am adding here just for consistency, but it sounds duplicated.

Also just realized that I missed the feature gate env vars. I will wait your feedback to make the changes, and not kill the existing jobs.

r4f4 replied:
If it doesn't exercise the capi provisioning, we don't need this in the __altinfra config. Now that capi is enabled in the "normal" installer, this config will probably be removed once we remove terraform from the Installer.

mtulio replied:
ACK, I will remove in the next review cycle.

mtulio commented:
Approval matrix:

| change | Team/owner | Status |
| -- | -- | -- |
| presubmit/periodic installer | installer | 🚧  Waiting for review |
| presubmit/periodic origin | TBD | 🚧 Waiting for review |
| periodics | TRT | 🚧 Waiting for review |
| [e2e workflow][e2e] | splat/cloud | 🚧 Waiting for review |
| [opct conformance workflow][opct] | splat | 🚧 Waiting for review |
| [IPOG platform external steps][ipog] | splat/cloud | 🚧 Waiting for review |

[e2e]: https://github.com/openshift/release/pull/45921/files#diff-ac552a6507719ccab58a6aa300dcc02d0f754e515081ce97f66dc897631b9cd4
[opct]: https://github.com/openshift/release/pull/45921/files#diff-17bbed49b9dcd358cf73de02a6ede4f21bbfd98603d81af400d7dcfa8ad4962f
[ipog]: https://github.com/openshift/release/pull/45921/files#diff-2346d9d60df807ca52aa3a330faeab731350e5e04442be54221dc41639d8c491

mtulio commented on ci-operator/step-registry/platform-external/cluster/aws/install/platform-external-cluster-aws-install-chain.yaml:
```diff
@@ -0,0 +1,9 @@
+  steps:
+  - ref: platform-external-cluster-aws-install-image
+  - ref: platform-external-cluster-aws-install
+  # WIP/experiment using single stack:
+  #- ref: platform-external-cluster-aws-install-stackset
```
TODO: remove in the next review round.

mtulio commented on ci-operator/step-registry/platform-external/cluster/aws/install/platform-external-cluster-aws-install-commands.sh:
```diff
@@ -0,0 +1,409 @@
+AWS_REGION=${LEASED_RESOURCE}
+export AWS_DEFAULT_REGION="${AWS_REGION}"  # CLI prefers the former
+
+# TODO(mtulio): move to step vars
+# export BOOTSTRAP_INSTANCE_TYPE=m6i.xlarge
+# export MASTER_INSTANCE_TYPE=m6i.xlarge
+# export WORKER_INSTANCE_TYPE=m6i.xlarge
+# export OCP_ARCH=amd64
```
Already moved, remove in the next review round.
```suggestion
```

mtulio commented on ci-operator/step-registry/platform-external/cluster/aws/install/platform-external-cluster-aws-install-commands.sh:
```diff
@@ -0,0 +1,409 @@
+
+GATHER_BOOTSTRAP_ARGS=
+NEW_STACKS="${SHARED_DIR}/aws_cfn_stacks"
+# mkdir -vp $NEW_STACKS
```
```suggestion
```

mtulio commented on ci-operator/step-registry/platform-external/cluster/aws/install/platform-external-cluster-aws-install-commands.sh:
```diff
@@ -0,0 +1,409 @@
+echo ${AWS_REGION} > ${SHARED_DIR}/AWS_REGION
+echo ${CLUSTER_NAME} > ${SHARED_DIR}/CLUSTER_NAME
+
+# TODO(mtulio): move to step vars or discovery from installer artifacts
+MACHINE_CIDR=10.0.0.0/16
+
+# begin bootstrapping
+# if openshift-install coreos print-stream-json 2> "${ARTIFACT_DIR}/err.txt" > /tmp/coreos.json; then
+#   RHCOS_AMI="$(jq -r --arg region "$AWS_REGION" '.architectures.x86_64.images.aws.regions[$region].image' /tmp/coreos.json)"
+#   # if [[ "${CLUSTER_TYPE}" == "aws-arm64" ]] || [[ "${OCP_ARCH}" == "arm64" ]]; then
+#   #   RHCOS_AMI="$(jq -r --arg region "$AWS_REGION" '.architectures.aarch64.images.aws.regions[$region].image' coreos.json)"
+#   # fi
+# else
+#   RHCOS_AMI="$(jq -r --arg region "$AWS_REGION" '.amis[$region].hvm' /var/lib/openshift-install/rhcos.json)"
+# fi
```
```suggestion
```

mtulio commented on ci-operator/step-registry/platform-external/pre/conf/manifests/platform-external-pre-conf-manifests-commands.sh:
```diff
@@ -0,0 +1,145 @@
+
+log "# << Saving to shared dir >> #"
+
+# cp -vf "${INSTALL_DIR}"/*.ign "${SHARED_DIR}"/
+# cp -vf "${INSTALL_DIR}"/auth/* "${SHARED_DIR}"/
+# cp -rvf "${INSTALL_DIR}"/auth "${SHARED_DIR}"/
+# cp -vf "${INSTALL_DIR}"/metadata.json "${SHARED_DIR}"/
```
```suggestion
```

mtulio commented on ci-operator/step-registry/platform-external/pre/conf/manifests/platform-external-pre-conf-manifests-ref.yaml:
```diff
@@ -0,0 +1,35 @@
+    default: ""
+    documentation: |-
+      When set to "yes" the CCM parameters will be set in the install-config.yaml.
+  # - name: BASE_DOMAIN
+  #   default: ""
+  #   documentation: |-
+  #     A fully-qualified domain or subdomain name. The base domain of the cloud provider is used for setting baseDomain variable of the install configuration of the cluster.
```
```suggestion
```

mtulio commented on ci-operator/step-registry/platform-external/pre/init/platform-external-pre-init-commands.sh:
```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
```
TODO review if we still need this step since those tools are available in `upi-installer` image. Are we using `upi-installer` across all required steps?

mtulio commented on ci-operator/step-registry/platform-external/cluster/wait-for/install-complete/platform-external-cluster-wait-for-install-complete-commands.sh:
```diff
@@ -0,0 +1,51 @@
+
+source "${SHARED_DIR}/init-fn.sh" || true
+
+# set +x
+# log "Waiting for install complete..."
+# openshift-install --dir=${INSTALL_DIR} wait-for install-complete 2>&1 | grep --line-buffered -v password &
+# wait "$!"
```
```suggestion
```

mtulio commented on ci-operator/step-registry/platform-external/cluster/aws/install/platform-external-cluster-aws-install-commands.sh:
```diff
@@ -0,0 +1,409 @@
+PRIVATE_SUBNETS="$(echo "${VPC_JSON}" | jq '.[] | select(.OutputKey == "PrivateSubnetIds").OutputValue')"  # explicitly keeping wrapping quotes
+PRIVATE_SUBNET_0="$(echo "${PRIVATE_SUBNETS}" | sed 's/"//g' | cut -d, -f1)"
+PRIVATE_SUBNET_1="$(echo "${PRIVATE_SUBNETS}" | sed 's/"//g' | cut -d, -f2)"
+# PRIVATE_SUBNET_2="$(echo "${PRIVATE_SUBNETS}" | sed 's/"//g' | cut -d, -f3)"
```
deploying into two zones only?

mtulio commented on ci-operator/step-registry/platform-external/cluster/aws/install/image/platform-external-cluster-aws-install-image-commands.sh:
```diff
@@ -0,0 +1,35 @@
+
+AWS_SHARED_CREDENTIALS_FILE=${CLUSTER_PROFILE_DIR}/.awscred
+export AWS_SHARED_CREDENTIALS_FILE
+#export OCP_ARCH=amd64
```
move to var
```suggestion
```

mtulio commented on ci-operator/step-registry/platform-external/ccm/conf/aws/platform-external-ccm-conf-aws-commands.sh:
```diff
@@ -0,0 +1,322 @@
+
+log "Creating CloudController Manager deployment"
+
+# TEMP reusing NS and RBAC from openshift-cloud-controller-manager:
```
```suggestion
# Create deployment manifest for AWS.
# TODO/FixMe(mtulio): create RBAC according to the upstream requirement. Currently it is using openshift-cloud-controller-manager (cluster defaults):
```

jcpowermac commented:
/lgtm

mtulio commented:
> Starting final tests:
> 
> /pj-rehearse more

Rehearsals finished successfully. o/

The `ci/rehearse/openshift/origin/master/e2e-external-aws*` failed as expected as we should use this presubmit (optional) to fix the existing E2E skips.

In the `periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws-ccm`  I am observing this e2e failing:

> [sig-arch] events should not repeat pathologically for ns/openshift-multus

Trying again to understand if it is flake or a test that need to be skipped at this time:

/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws-ccm

mtulio commented:
> > Starting final tests:
> > /pj-rehearse more
> 
> Rehearsals finished successfully. o/
> 
> The `ci/rehearse/openshift/origin/master/e2e-external-aws*` failed as expected as we should use this presubmit (optional) to fix the existing E2E skips.
> 
> In the `periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws-ccm` I am observing this e2e failing:
> 
> > [sig-arch] events should not repeat pathologically for ns/openshift-multus
> 
> Trying again to understand if it is flake or a test that need to be skipped at this time:
> 
> /pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws-ccm

Rehearsal finished successfully, indicating the last run was flake.

mtulio commented on ci-operator/jobs/openshift/origin/openshift-origin-master-presubmits.yaml:
```diff
@@ -807,8 +807,6 @@ presubmits:
     cluster: build05
     context: ci/prow/e2e-aws-ovn-edge-zones
     decorate: true
-    decoration_config:
-      timeout: 6h0m0s
```
unexpected change due line removal on the job definition. must be reverted.

mtulio commented:
Addressing PR review to remove from `altinfra` and clean up code comments. Nothing that will impact the previously rehearsals. I will run couple before ACK.

/pj-rehearse periodic-ci-openshift-release-master-ci-4.16-e2e-external-aws-ccm periodic-ci-openshift-release-master-nightly-4.16-opct-external-aws-ccm	 pull-ci-openshift-installer-master-e2e-external-aws-ccm

r4f4 reviewed (commented):
/approve

From the Installer side, the presubmits are optional and do not run by default, so pretty harmless.

You did mention they are based on the UPI workflows, so a nitpick from me would be to reflect that on the job's name? so something like: `e2e-external-aws-ovn-upi`

mtulio commented:
> /approve
> 
> From the Installer side, the presubmits are optional and do not run by default, so pretty harmless.

Thanks!

> You did mention they are based on the UPI workflows, so a nitpick from me would be to reflect that on the job's name? so something like: `e2e-external-aws-ovn-upi`

Good point. It is based in UPI, but past discussions in the WG, they all agreed-ed to prevent using the "UPI" terminology for platform External clusters as it could cause some confusion with existing install methods. Currently all installs from platform external supports only user/partner-provided infrastructure, so having in the name, at least for now, would be redundant.

mtulio commented:
> Needs approval from an approver in each of these files:
> * **[ci-operator/config/openshift/origin/OWNERS](https://github.com/openshift/release/blob/master/ci-operator/config/openshift/origin/OWNERS)**
> * **[ci-operator/config/openshift/release/OWNERS](https://github.com/openshift/release/blob/master/ci-operator/config/openshift/release/OWNERS)**
> * **[ci-operator/jobs/openshift/origin/OWNERS](https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/origin/OWNERS)**
> * **[ci-operator/jobs/openshift/release/OWNERS](https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/release/OWNERS)**
> * **[ci-operator/step-registry/openshift/OWNERS](https://github.com/openshift/release/blob/master/ci-operator/step-registry/openshift/OWNERS)**

Hi Devan and Stephan, would you mind taking a look at those new periodics and presubmit on origin? The PR description, section "Approval matrix:" has more details. Thanks

/assign @dgoodwin @stbenjam

mtulio commented:
trying again :)
/assign stbenjam

mtulio commented:
Job [1790769039983775744][1790769039983775744] `openshift/installer/master/e2e-external-aws-ccm` failed the test `[sig-arch] ClusterOperators [apigroup:config.openshift.io] should define at least one namespace in their lists of related objects [Suite:openshift/conformance/parallel] `:

```
  FAIL! -- 0 Passed | 1 Failed | 0 Pending | 0 Skipped
fail [github.com/openshift/origin/test/extended/operators/clusteroperators.go:58]: ClusterOperator: control-plane-machine-set
```

It seems to be a flake, but should CPMS be active on non-integrated platforms even if we are removing related CPMS manifests in the [manifest stage][manifest]?! (open questions) cc @elmiko 

Re-testing:

/pj-rehearse pull-ci-openshift-origin-master-e2e-external-aws


[manifest]: https://github.com/openshift/release/pull/45921/files#diff-c0c19e3f37fa8bb49e5d54f8b7d72180835aeb2a96a474245b4f94d3e8ef8563R123-R130
[1790769039983775744]: https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/45921/rehearse-45921-pull-ci-openshift-installer-master-e2e-external-aws-ccm/1790769039983775744

mtulio commented:
Correct job:

/pj-rehearse pull-ci-openshift-origin-master-e2e-external-aws-ccm

Canceling the triggered in my last comment (without CCM)

mtulio commented:
> Correct job:
> 
> 
> 
> /pj-rehearse pull-ci-openshift-origin-master-e2e-external-aws-ccm
> 
> 
> 
> Canceling the triggered in my last comment (without CCM)

/pj-rehearse pull-ci-openshift-installer-master-e2e-external-aws-ccm

mtulio commented:
All required rehearsals are passing. The failing one (origin) is expected due the enforcement of cleaning e2e skips:

/pj-rehearse ack

mtulio commented:
Hey @elmiko , would you mind taking a look here related to [SPLAT-1158](https://issues.redhat.com/browse/SPLAT-1158), please? Thanks

jcpowermac commented:
/lgtm

mtulio commented:
Previously runs already ACK. Rebase removed the labels.
/pj-rehearse ack

mtulio commented:
Adding 4.17 as requested in Slack.

/pj-rehearse periodic-ci-openshift-release-master-nightly-4.17-e2e-external-aws-ccm

This PR is ready for review, I will hold only to run the new job:
/hold

jcpowermac commented:
/lgtm

bradmwilliams commented:
/approve

elmiko reviewed (commented):
this looks mostly good to me, i would like to study a little more, but don't hold on my part.

mtulio commented:
> Adding 4.17 as requested in Slack.
> 
> /pj-rehearse periodic-ci-openshift-release-master-nightly-4.17-e2e-external-aws-ccm
> 
> This PR is ready for review, I will hold only to run the new job: /hold

/pj-rehearse ack
/unhold

stbenjam commented:
/approve
